module/icp/asm-x86_64/modes/aesni-gcm-x86_64.S

```diff
@@ -0,0 +1,892 @@
+# Copyright 2013-2016 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+#
+# ====================================================================
+# Written by Andy Polyakov <appro@openssl.org> for the OpenSSL
+# project. The module is, however, dual licensed under OpenSSL and
+# CRYPTOGAMS licenses depending on where you obtain it. For further
+# details see http://www.openssl.org/~appro/cryptogams/.
+# ====================================================================
```

Comment:
Please add the gcm and ghash implementations to the top level `COPYRIGHT` file along with the other exceptions.  You'll want to also add a `THIRDPARTYLICENSE` file for the correct license a well.

Reply:
Will do.

Reply:
Done, a337b63.